### PR TITLE
Azure storage diagnostic setting

### PR DESCRIPTION
**Pull request by laukik-uptycs**
### Description
- azure storage diagnostic setting table added
- table config is modified to accommodate the new table
- plugins.go is modified to accommodate the new table

### References
- https://pkg.go.dev/github.com/Azure/azure-sdk-for-go@v59.4.0+incompatible/services/preview/monitor/mgmt/2021-04-01-preview/insights#DiagnosticSettingsCategoryClient.List

### Testing
- tested using OSqueryI

### CODE REVIEW

File: extension/azure/storage/azure_storage_diagnostic_setting.go [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+
+// StorageDiagnosticSettingsColumns returns the list of columns in the table
+func StorageDiagnosticSettingColumns() []table.ColumnDefinition {
+	return []table.ColumnDefinition{
+		table.TextColumn("id"),
+		table.TextColumn("name"),
+		table.TextColumn("storageAccountId"),
+		table.TextColumn("type"),
+		table.TextColumn("propertise_log"),
+		table.TextColumn("propertise_metrics"),
```
Comment (alende-uptycs) [inline extension/azure/storage/azure_storage_diagnostic_setting.go:30]: properties

File: extension/azure/storage/azure_storage_diagnostic_setting.go [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+			utilities.GetLogger().WithFields(log.Fields{
+				"tableName":     storageAccount,
+				"resourceGroup": rg,
+				"errString":     err.Error(),
+			}).Error("failed to get resource list")
+			continue
+		}
+
+		resource := resourceItr.Value()
+		diagnosticSetting, err := getStorageDiagnosticSetting(session, rg, wg, resultMap, tableConfig, *resource.ID)
```
Comment (alende-uptycs) [inline extension/azure/storage/azure_storage_diagnostic_setting.go:129]: We need to fetch diagnostic settings for file, blob, queue and table services also.

File: extension/azure/storage/table_config.json [hunk 3]
```diff
@@ -2065,5 +2065,164 @@
...
+              "sourceName": "propertise_metrics",
+              "targetName": "propertise_metrics",
+              "targetType": "TEXT",
+              "enabled": true
+            },
+            {
+              "sourceName": "eventHubAuthorizationRuleId",
+              "targetName": "event_hub_authorization_rule_id",
+              "targetType": "TEXT",
+              "enabled": false
```
Comment (alende-uptycs) [inline extension/azure/storage/table_config.json:2116]: enable

File: extension/azure/storage/table_config.json [hunk 4]
```diff
@@ -2065,5 +2065,164 @@
...
+            },
+            {
+              "sourceName": "eventHubName",
+              "targetName": "event_hub_name",
+              "targetType": "TEXT",
+              "enabled": false
```
Comment (alende-uptycs) [inline extension/azure/storage/table_config.json:2122]: enable

File: extension/azure/storage/table_config.json [hunk 5]
```diff
@@ -2065,5 +2065,164 @@
...
+            },
+            {
+              "sourceName": "logAnalyticsDestinationType",
+              "targetName": "log_analytics_destination_type",
+              "targetType": "TEXT",
+              "enabled": false
```
Comment (alende-uptycs) [inline extension/azure/storage/table_config.json:2128]: enable

File: extension/azure/storage/table_config.json [hunk 6]
```diff
@@ -2065,5 +2065,164 @@
...
+            },
+            {
+              "sourceName": "logs",
+              "targetName": "logs",
+              "targetType": "TEXT",
+              "enabled": false
```
Comment (alende-uptycs) [inline extension/azure/storage/table_config.json:2134]: enable

File: extension/azure/storage/table_config.json [hunk 7]
```diff
@@ -2065,5 +2065,164 @@
...
+              "sourceName": "logs_retentionPolicy_enabled",
+              "targetName": "logs_retention_policy_enabled",
+              "targetType": "TEXT",
+              "enabled": false
+            },
+            {
+              "sourceName": "metrics",
+              "targetName": "metrics",
+              "targetType": "TEXT",
+              "enabled": false
```
Comment (alende-uptycs) [inline extension/azure/storage/table_config.json:2170]: enable

File: extension/azure/storage/table_config.json [hunk 8]
```diff
@@ -2065,5 +2065,164 @@
...
+              "sourceName": "metrics_timeGrain",
+              "targetName": "metrics_time_grain",
+              "targetType": "TEXT",
+              "enabled": false
+            },
+            {
+              "sourceName": "serviceBusRuleId",
+              "targetName": "service_bus_rule_id",
+              "targetType": "TEXT",
+              "enabled": false
```
Comment (alende-uptycs) [inline extension/azure/storage/table_config.json:2212]: enable

File: extension/azure/storage/table_config.json [hunk 9]
```diff
@@ -2065,5 +2065,164 @@
...
+            },
+            {
+              "sourceName": "storageAccountId",
+              "targetName": "storage_account_id",
+              "targetType": "TEXT",
+              "enabled": false
```
Comment (alende-uptycs) [inline extension/azure/storage/table_config.json:2218]: enable

File: extension/azure/storage/table_config.json [hunk 10]
```diff
@@ -2065,5 +2065,164 @@
...
+            },
+            {
+              "sourceName": "workspaceId",
+              "targetName": "workspace_id",
+              "targetType": "TEXT",
+              "enabled": false
```
Comment (alende-uptycs) [inline extension/azure/storage/table_config.json:2224]: enable

File: extension/azure/storage/azure_storage_diagnostic_setting.go (hunk 2 above)
Comment (laukik-uptycs) [reply extension/azure/storage/azure_storage_diagnostic_setting.go:129]: Ok

**Review by alende-uptycs — DISMISSED** [review]

**Review by ssuthar-uptycs — APPROVED** [review]

**Pull request merged**